src/js/components/Menu/Menu.js

```diff
@@ -236,7 +237,7 @@ const Menu = forwardRef((props, ref) => {
           // make it accessible at the end of all menu items
           buttonRefs[items.length] = r;
         }}
-        a11yTitle={a11yTitle || messages.closeMenu || 'Close Menu'}
+        a11yTitle={ariaLabel || a11yTitle || messages.closeMenu || 'Close Menu'}
```

Comment:
I think that with your recent Button fix, the `aria-label` would work automatically, so I don't think we need to explicitly handle it.